Sources/OTel/Metrics/OTelMetricReader/OTelPeriodicExportingMetricsReader.swift

```diff
@@ -0,0 +1,92 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift OTel open source project
+//
+// Copyright (c) 2024 Moritz Lang and the Swift OTel project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import AsyncAlgorithms
+import Logging
+import ServiceLifecycle
+
+@_spi(Metrics)
+public struct OTelPeriodicExportingMetricsReader<Clock: _Concurrency.Clock>: Service, CustomStringConvertible where Clock.Duration == Duration {
+    public let description = "OTelPeriodicExportingMetricsReader"
+
+    private let logger = Logger(label: "OTelPeriodicExportingMetricsReader")
+
+    var resource: OTelResource
+    var producer: OTelMetricProducer // TODO: support for multiple producers?
+    var exporter: OTelMetricExporter
+    var configuration: OTelPeriodicExportingMetricsReaderConfiguration
+    var clock: Clock
+
+    init(
+        resource: OTelResource,
+        producer: OTelMetricProducer,
+        exporter: OTelMetricExporter,
+        configuration: OTelPeriodicExportingMetricsReaderConfiguration,
+        clock: Clock
+    ) {
+        self.resource = resource
+        self.producer = producer
+        self.exporter = exporter
+        self.configuration = configuration
+        self.clock = clock
+    }
+
+    func tick() async {
+        logger.trace("Reading metrics from producer.")
+        let metrics = producer.produce()
+        let batch = [
+            OTelResourceMetrics(
+                resource: resource,
+                scopeMetrics: [OTelScopeMetrics(
+                    scope: .init(name: "swift-otel", version: OTelLibrary.version, attributes: [], droppedAttributeCount: 0),
+                    metrics: metrics
+                )]
+            ),
+        ]
+        logger.debug("Exporting metrics.", metadata: ["count": "\(metrics.count)"])
```

Comment:
In `OTelBatchSpanProcessor`, I'm creating an `exportLogger` to tie together the pre-export log and post-export success/failure statements with a batch-id. Do you think something similar could be useful here too?

https://github.com/slashmo/swift-otel/blob/a22da19099cbe332381aa9afb9cf3266590ce953/Sources/OTel/Tracing/Processing/Batch/OTelBatchSpanProcessor.swift#L124

Reply:
We could do that, but I'd imagine it be pretty easy to correlate these in the logs since the timeout will always be less than the interval.